File: backend/db/seeds.rb
if User.where(role: 'admin').where(email: 'insanemito@gmail.com').count == 0
  admin_email = 'insanemito@gmail.com'
  admin_username = 'insanemito'

  puts "Creating default admin user..."
  
  begin
    admin = User.find_or_initialize_by(email: admin_email)
    
    # Check if ADMIN_PASSWORD is set in environment
    if ENV['ADMIN_PASSWORD'].present?
      admin_password = ENV['ADMIN_PASSWORD']
      admin.assign_attributes(
        password: admin_password,
        password_confirmation: admin_password,
        username: admin_username,
        role: 'admin',
        provider: nil, # allow manual login
        uid: nil
      )
    else
      # In development/test, create without password for OAuth-only access
      # Or require password to be set via environment variable
      if Rails.env.production?
        puts "❌ ADMIN_PASSWORD environment variable is required in production!"
        puts "Set it with: export ADMIN_PASSWORD='your_secure_password'"
        exit 1
      else
        # Development: Create user without password (OAuth only) or generate random one
        temp_password = SecureRandom.alphanumeric(16)
        admin.assign_attributes(
          password: temp_password,
          password_confirmation: temp_password,
          username: admin_username,
          role: 'admin',
          provider: nil,
          uid: nil
        )
        puts "🔐 Generated temporary password for development: #{temp_password}"
        puts "💡 Or just use Google OAuth to sign in!"
      end
    end
    
    admin.save!
    
    puts "✅ Admin user created or updated: #{admin.email} (username: #{admin.username})"
    
  rescue ActiveRecord::RecordInvalid => e
    puts "❌ Failed to create admin user: #{e.message}"
    puts "Please check your User model validations and try again."
    exit 1
  rescue => e
    puts "❌ Unexpected error creating admin user: #{e.message}"
    exit 1
  end
else
  admin_count = User.where(role: 'admin').count
  puts "Admin user already exists for insanemito@gmail.com, skipping creation."
end

# Add other seed data below as needed
# Example: Create default genres, playlists, etc.

puts "\n🌱 Seeding completed successfully!"

File: backend/seeds.js
const path = require('path');
require('dotenv').config({ path: path.resolve(__dirname, '..', '.env') });

const mongoose = require('mongoose');
const User = require('./models/User');

async function seed() {
  try {
    const uri = process.env.MONGODB_URI || 'mongodb://localhost:27017/music_player_development';
    await mongoose.connect(uri);
    console.log('Connected to MongoDB for seeding');

    const adminEmail = 'insanemito@gmail.com';
    const adminUsername = 'insanemito';

    const existingAdmin = await User.findOne({ email: adminEmail, role: 'admin' });
    if (existingAdmin) {
      console.log('Admin user already exists for insanemito@gmail.com, skipping creation.');
    } else {
      const adminPassword = process.env.ADMIN_PASSWORD;
      if (!adminPassword && process.env.NODE_ENV === 'production') {
        console.error('ADMIN_PASSWORD environment variable is required in production!');
        process.exit(1);
      }

      const password = adminPassword || `temp_${Math.random().toString(36).slice(2, 18)}`;

      let admin = await User.findOne({ email: adminEmail });
      if (admin) {
        admin.role = 'admin';
        admin.passwordHash = password;
        if (admin.profile) {
          admin.profile.username = adminUsername;
        } else {
          admin.profile = { username: adminUsername };
        }
        await admin.save();
      } else {
        admin = new User({
          email: adminEmail,
          passwordHash: password,
          role: 'admin',
          profile: { username: adminUsername }
        });
        await admin.save();
      }

      console.log(`Admin user created or updated: ${admin.email} (username: ${adminUsername})`);
      if (!adminPassword) {
        console.log(`Generated temporary password for development: ${password}`);
      }
    }

    console.log('\nSeeding completed successfully!');
  } catch (err) {
    console.error('Seeding error:', err.message);
    process.exit(1);
  } finally {
    await mongoose.disconnect();
  }
}

seed();
